db_stress_tool/db_stress_test_base.cc

```diff
@@ -307,7 +307,20 @@ void StressTest::FinishInitDb(SharedState* shared) {
     fprintf(stdout, "Compaction filter factory: %s\n",
             compaction_filter_factory->Name());
   }
-  // TODO(ajkr): First restore if there's already a trace.
+
+  if (shared->HasHistory() && IsStateTracked()) {
```

Comment:
This `IsStateTracked()` causes a problem in the following sequence:

- `-sync_fault_injection=1 -test_batches_snapshot=0`
- `-test_batches_snapshot=1`
- `-test_batches_snapshot=0`

The second command can cause DB seqno to advance beyond what is recoverable in expected values, which causes the third command to fail.